srt-protocol/src/connection/mod.rs

```diff
@@ -23,13 +23,13 @@ use crate::{
     statistics::SocketStatistics,
 };
 
-#[derive(Debug)]
+#[derive(Debug, Eq, PartialEq)]
```

Comment:
I don't disagree it might seem strange. I probably was forced to add this because this data structure ended up in a enum that derived Eq. But... in general I like the values in code (i.e. "Whole Values"/"Value Objects"/"Value Types") to all be equatable buy default. Settings and configuration are values too. This makes unit tests much more easy and ergonomic. If there was an alternative way to get easy deep structural comparisons in test, I'd be open to using that instead. The ability to automagically get pretty printing (Debug) and structural equality (Eq, PartialEq) are two big reasons I love Rust. The standard library has equatable collections, HashMaps, etc. mostly to enable this powerful feature. I've been spoiled before by using typed functional languages before, and got accustom to this style. Honestly, I'd prefer if Rust forced us to opt out of structural equality and pretty printing, for the occasional large stateful "object" with no legitimate approximation of value equality, or when a debug representation makes no sense. Even better though would be auto-derived structural diffing functionality, that's really what I want. Don't just test for equality, show me exactly what isn't the same. :)